identity/src/main/java/com/stripe/android/identity/navigation/NavigationDestinations.kt

```diff
@@ -0,0 +1,237 @@
+        ARG_CAUSE to cause
+    )
+
+    internal companion object {
+        const val ERROR = "Error"
+        const val ARG_ERROR_TITLE = "errorTitle"
+        const val ARG_ERROR_CONTENT = "errorContent"
+
+        // if set, shows go_back button, clicking it would navigate to the destination.
+        const val ARG_GO_BACK_BUTTON_TEXT = "goBackButtonText"
+        const val ARG_GO_BACK_BUTTON_DESTINATION = "goBackButtonDestination"
+
+        // if set to true, clicking bottom button and pressBack would end flow with Failed
+        const val ARG_SHOULD_FAIL = "shouldFail"
+        const val ARG_CAUSE = "cause"
+
+        // Indicates the server returns a requirementError that doesn't match with current Fragment.
+        //  E.g ConsentFragment->DocSelectFragment could only have BIOMETRICCONSENT error but not IDDOCUMENTFRONT error.
+        // If this happens, set the back button destination to [DEFAULT_BACK_BUTTON_DESTINATION]
+        const val UNEXPECTED_DESTINATION = -1
+
+        val DEFAULT_BACK_BUTTON_NAVIGATION =
+            R.id.action_errorFragment_to_consentFragment
+
+        val Route = object : DestinationRoute() {
+            override val routeBase = ERROR
+            override val arguments = listOf(
+                navArgument(ARG_ERROR_TITLE) {
+                    type = NavType.StringType
+                },
+                navArgument(ARG_ERROR_CONTENT) {
+                    type = NavType.StringType
+                },
+                navArgument(ARG_GO_BACK_BUTTON_TEXT) {
+                    type = NavType.StringType
+                },
+                navArgument(ARG_GO_BACK_BUTTON_DESTINATION) {
+                    type = NavType.StringType
+                },
+                navArgument(ARG_SHOULD_FAIL) {
+                    type = NavType.BoolType
+                }
+            )
+        }
+    }
+}
+
+internal object ConfirmationDestination : IdentityTopLevelDestination() {
+    private const val CONFIRMATION = "Confirmation"
+    override val destination = R.id.action_global_confirmationFragment
+    override val destinationRoute = object : DestinationRoute() {
+        override val routeBase = CONFIRMATION
+    }
+    override val routeWithArgs = destinationRoute.route
+}
+
+internal object SelfieDestination : IdentityTopLevelDestination() {
+    private const val SELFIE = "Selfie"
+    override val destination = R.id.action_global_selfieFragment
+    override val destinationRoute = object : DestinationRoute() {
+        override val routeBase = SELFIE
+    }
+    override val routeWithArgs = destinationRoute.route
+}
+
+/**
+ * Navigate to the [IdentityTopLevelDestination] by calling NavController.navigate(fragmentID, bundle).
+ * TODO(ccen) change its implementation to NavController.navigate(destination.route) when all fragments are moved.
+ */
+internal fun NavController.navigateTo(destination: IdentityTopLevelDestination) {
+//    navigate(destination.routeWithArgs)
```

Comment:
This call will be used after `composable`s are created within `IdentityActivity`